### PR TITLE
feat(gaussdb): change auditlog to golangsdk

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
1. change auditlog API  to golangsdk
2. change the endpoint of gaussdb,  #https://github.com/huaweicloud/terraform-provider-huaweicloud/pull/2123#issuecomment-1112112380

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccServiceEndpoints_Database'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccServiceEndpoints_Database -timeout 360m -parallel 4
=== RUN   TestAccServiceEndpoints_Database
--- PASS: TestAccServiceEndpoints_Database (1.29s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       1.404s
```

```

make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccGaussDBInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccGaussDBInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussDBInstance_basic
=== PAUSE TestAccGaussDBInstance_basic
=== CONT  TestAccGaussDBInstance_basic
--- PASS: TestAccGaussDBInstance_basic (1229.88s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       1230.007s
```

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_gaussdb_mysql_instance.go [hunk 1]
```diff
@@ -344,14 +343,9 @@ func resourceGaussDBInstanceCreate(d *schema.ResourceData, meta interface{}) err
 	if err != nil {
 		return fmtp.Errorf("error creating HuaweiCloud GaussDB client: %s ", err)
 	}
-	tagClient, err := config.GaussdbV3TagClient(GetRegion(d, config))
 	if err != nil {
 		return fmtp.Errorf("error creating GaussDB client: %s ", err)
 	}
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_gaussdb_mysql_instance.go]: These lines should also be removed

File: huaweicloud/resource_huaweicloud_gaussdb_mysql_instance.go [hunk 2]
```diff
@@ -582,14 +576,9 @@ func resourceGaussDBInstanceRead(d *schema.ResourceData, meta interface{}) error
 	if err != nil {
 		return fmtp.Errorf("error creating HuaweiCloud GaussDB client: %s", err)
 	}
-	tagClient, err := config.GaussdbV3TagClient(GetRegion(d, config))
 	if err != nil {
 		return fmtp.Errorf("error creating GaussDB client: %s ", err)
 	}
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_gaussdb_mysql_instance.go]: ditto

File: huaweicloud/resource_huaweicloud_gaussdb_mysql_instance.go [hunk 3]
```diff
@@ -762,18 +748,13 @@ func resourceGaussDBInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 	if err != nil {
 		return fmtp.Errorf("error creating HuaweiCloud GaussDB client: %s ", err)
 	}
-	tagClient, err := config.GaussdbV3TagClient(GetRegion(d, config))
 	if err != nil {
 		return fmtp.Errorf("error creating GaussDB client: %s ", err)
 	}
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_gaussdb_mysql_instance.go]: ditto

**Comment by hmbig2** [comment]
@ShiChangkuo  remove the dulp

**Pull request merged**